src/backend/distributed/deparser/citus_ruleutils.c

```diff
@@ -684,12 +684,13 @@ deparse_shard_index_statement(IndexStmt *origStmt, Oid distrelid, int64 shardid,
 							  StringInfo buffer)
 {
 	IndexStmt *indexStmt = copyObject(origStmt); /* copy to avoid modifications */
-	char *relationName = indexStmt->relation->relname;
-	char *indexName = indexStmt->idxname;
 
 	/* extend relation and index name using shard identifier */
-	AppendShardIdToName(&relationName, shardid);
-	AppendShardIdToName(&indexName, shardid);
+	AppendShardIdToName(&(indexStmt->relation->relname), shardid);
```

Comment:
a personal taste, I prefer to see this a little more explicit, feel free to ignore:
```C
--- a/src/backend/distributed/deparser/citus_ruleutils.c
+++ b/src/backend/distributed/deparser/citus_ruleutils.c
@@ -683,26 +683,27 @@ void
 deparse_shard_index_statement(IndexStmt *origStmt, Oid distrelid, int64 shardid,
                                                          StringInfo buffer)
 {
-       IndexStmt *indexStmt = copyObject(origStmt); /* copy to avoid modifications */
+       /* we need to copy because we are modifying the origStmt */
+       IndexStmt *indexStmt = copyObject(origStmt);
 
-       /* extend relation and index name using shard identifier */
-       AppendShardIdToName(&(indexStmt->relation->relname), shardid);
-       AppendShardIdToName(&(indexStmt->idxname), shardid);
+       char **relationName = &(indexStmt->relation->relname);
+       char **indexName = &(indexStmt->idxname);
 
-       char *relationName = indexStmt->relation->relname;
-       char *indexName = indexStmt->idxname;
+       /* extend relation and index name using shard identifier on the indexStmt */
+       AppendShardIdToName(relationName, shardid);
+       AppendShardIdToName(indexName, shardid);
 
        /* use extended shard name and transformed stmt for deparsing */
-       List *deparseContext = deparse_context_for(relationName, distrelid);
+       List *deparseContext = deparse_context_for(*relationName, distrelid);
        indexStmt = transformIndexStmt(distrelid, indexStmt, NULL);
 
        appendStringInfo(buffer, "CREATE %s INDEX %s %s %s ON %s USING %s ",
                                         (indexStmt->unique ? "UNIQUE" : ""),
                                         (indexStmt->concurrent ? "CONCURRENTLY" : ""),
                                         (indexStmt->if_not_exists ? "IF NOT EXISTS" : ""),
-                                        quote_identifier(indexName),
+                                        quote_identifier(*indexName),
                                         quote_qualified_identifier(indexStmt->relation->schemaname,
-                                                                                               relationName),
+                                                                                               *relationName),
                                         indexStmt->accessMethod);
```

Reply:
I'm slightly more inclined to make this pr as small as possible, so merging as is